generate_endf.py

```diff
@@ -0,0 +1,358 @@
+
+# ==============================================================================
+# EXTRACT FILES FROM TGZ
+
+if args.extract:
+    for particle in args.particles:
+
+        if release_details[args.release][particle]['file_type'] == 'wmp':
+            extraction_dir = args.destination / 'wmp' / particle
+        elif release_details[args.release][particle]['file_type'] == 'endf':
+            extraction_dir = endf_files_dir / particle
+        Path.mkdir(extraction_dir, parents=True, exist_ok=True)
+
+        for f in release_details[args.release][particle]['compressed_files']:
+            fname = Path(f).name
+            # Extract files different depending on compression method
+            if fname.endswith('.zip'):
+                print(f'Extracting {fname}...')
+                with zipfile.ZipFile(download_path / particle / fname) as zipf:
+                    # Extracts files without folder structure in the zip file
+                    for member in zipf.namelist():
+                        filename = Path(member).name
+                        # skip directories
+                        if not filename:
+                            continue
+                        source = zipf.open(member)
+                        target = open(extraction_dir / filename, "wb")
+                        with source, target:
+                            copyfileobj(source, target)
+            elif fname.endswith('.tar.gz'):
+                with tarfile.open(download_path / particle / fname, 'r') as tgz:
+                    print(f'Extracting {fname}...')
+                    # extract files ignoring the internal folder structure
+                    for member in tgz.getmembers():
+                        if member.isreg():
+                            member.name = Path(member.name).name
+                            tgz.extract(member, path=extraction_dir)
+            else:
+                # File is not compressed. Used for erratafiles. This ensures
+                # the n-005_B_010.endf erratafile overwrites the orginal
+                copy(download_path/particle/fname, extraction_dir/fname)
+
+    if args.cleanup and download_path.exists():
+        rmtree(download_path)
+
+# =========================================================================
+# PROCESS INCIDENT NEUTRON AND THERMAL SCATTERING DATA IN PARALLEL
+
+# Create output directory if it doesn't exist
+for particle in args.particles:
+    particle_destination = args.destination / particle
+    particle_destination.mkdir(parents=True, exist_ok=True)
+
+library = openmc.data.DataLibrary()
+
+if 'neutron' in args.particles:
+    particle = 'neutron'
+    with Pool() as pool:
+        details = release_details[args.release][particle]
+        results = []
+        for filename in details['endf_files']:
+
+            # Skip neutron evaluation that fails the processing stage
+            if filename == 'n-000_n_001.endf' and args.release == 'b8.0':
```

Comment:
Just noticed this doesn't quite work because `filename` is the full path. This should though:
```suggestion
            if filename.name == 'n-000_n_001.endf':
```
Note that I've removed the extra check for 8.0 because even for future releases we won't want the n-n evaluation to be processed.

Reply:
Thanks Paul, this script takes so long to run on my computer it is a bit of a pain to test. Sorry for missing that